tests/cephadm/test_config.py

```diff
@@ -0,0 +1,62 @@
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class CephadmOpsFailure(Exception):
+    pass
+
+
+def set_multiple_public_network(cls):
+    """
+    To set multiple public network to ceph.
+    Args:
+        cls: cephadm instance object
+
+    Returns:
+        False -> Failed to set multiple public network to ceph
+        True -> Test passed to set multiple public network to ceph
+    """
+    out, _ = cls.shell(args=["ceph", "config", "get", "mon", "public_network"])
+    ceph_pub_network = out.rstrip()
+    mask = ceph_pub_network.rsplit(sep="/")
+    # logic to create new public ip by incrementing mask count
+    new_mask = int(mask[-1]) + 1
+    new_pub_network = f"{mask[0]}/{str(new_mask)}"
+    multiple_network = f"{ceph_pub_network},{new_pub_network}"
+    cmd = ["ceph", "config", "set", "mon", "public_network", multiple_network]
+    try:
+        out, err = cls.shell(args=cmd)
+        log.debug("Set multiple public network to ceph mon successfully")
+    except Exception:
+        log.error("Failed to set multiple public network to ceph")
+        raise CephadmOpsFailure(traceback.format_exc)
+
+    # To validate the set address using get operation
+    public_network, _ = cls.shell(
+        args=["ceph", "config", "get", "mon", "public_network"]
+    )
+    if public_network.rstrip() == multiple_network:
+        return True
+    return False
+
+
+def run(ceph_cluster, **kw):
```

Comment:
To test this case, I think we need ceph nodes to be in different networks.
- Please check is there any possiblity in PSI. because In IBM we have only one network in VPC.
- Please check if you want this to be executed in IBM test pipeline or could cover this in PSI.

Reply:
This test for cephadm-specific CLI functionality not the multiple public network functionality.
Hence no need of real different public network infra.